static/js/components/StarList.jsx

```diff
@@ -21,25 +38,49 @@ const StarList = ({ sourceId }) => {
     fetchStarList();
   }, [sourceId, dispatch]);
 
-  return (
-    <div className={styles.starListDiv}>
-      <code className={styles.starList}>
-        {
-          starList && starList.map((item, idx) => (
-            // eslint-disable-next-line react/no-array-index-key
-            <React.Fragment key={idx}>
-              { item.str }
-              <br />
-            </React.Fragment>
-          ))
+  return <StarListBody starList={starList} />;
+};
+
+export const ObservingRunStarList = () => {
+  const [starList, setStarList] = useState([{ str: 'Loading starlist...' }]);
+  const { assignments } = useSelector((state) => state.observingRun);
+  const dispatch = useDispatch();
+
+  const fetchStarList = async () => {
+    const promises = assignments.map(
+      (assignment) => (
+        dispatch(
+          GET(
+            `/api/sources/${assignment.obj_id}/offsets?facility=Keck`, 'skyportal/FETCH_STARLIST'
+          )
+        )
+      )
+    );
+
+    const starlistInfo = [];
+    await Promise.allSettled(promises)
```

Comment:
I don't think this is quite right; I think you should be using either `await` or `promise.then()` here, but not both. 
```suggestion
    values = await Promise.allSettled(promises);
    values.forEach(...);
```

Reply:
fixed